File: Java/labs/lab22_Template_Method/task_2_1 – solution/src/MeatFeastPizza.java
public class MeatFeastPizza extends Pizza{

  @Override
  public void cook() {
    System.out.println("Cook in the oven for 20 minutes");
  }

  public void addToppings() {
    super.addToppings();
    System.out.println("Add pepperoni");
    System.out.println("Add ham");
    System.out.println("Add chicken");
  }

}

File: Cpp/lab22_Template_Method/task_2_1 - solution/src/meat_feast_pizza.cpp
#include <iostream>

#include "meat_feast_pizza.hpp"


void MeatFeastPizza::addToppings()
{
  this->Pizza::addToppings();
  std::cout << "Add pepperoni" << std::endl
            << "Add ham" << std::endl
            << "Add chicken" << std::endl;
}

void MeatFeastPizza::cook()
{
    std::cout << "Cook in the oven for 20 minutes" << std::endl;
}
